fhir-search/src/main/java/com/ibm/fhir/search/util/SearchUtil.java

```diff
@@ -801,9 +791,7 @@ public static FHIRSearchContext parseQueryParameters(Class<?> resourceType,
                 if (isSearchResultParameter(name)) {
                     parseSearchResultParameter(resourceType, context, name, params, lenient);
                     // _include and _revinclude parameters cannot be mixed with _summary=text
-                    // TODO: this will fire on each search result parameter; maybe move this above to where we handle _sort + _include/_revinclude?
-                    if (context.getSummaryParameter() != null
-                            && context.getSummaryParameter().equals(SummaryValueSet.TEXT)) {
+                    if (SummaryValueSet.TEXT.equals(context.getSummaryParameter())) {
```

Comment:
Seems like this check should be moved to the bottom after everything has been parsed, maybe before the check on line 925. Should also probably add a log message if we're removing parameters from the search.

Reply:
Moved to top. Treat as error if strict, but log message and handle as before if lenient.